mainwindow.cpp

```diff
@@ -43,6 +41,21 @@ MainWindow::MainWindow(QWidget *parent, const char *config_file)
 		QMessageBox::about(this, "About LabRecorder", infostr);
 	});
 
+	// Wheenver locationEdit is changed, print the final result.
+	connect(ui->locationEdit, &QLineEdit::textChanged, this, &MainWindow::printReplacedFilename);
+	connect(ui->experimentNumberSpin, QOverload<int>::of(&QSpinBox::valueChanged), this, &MainWindow::printReplacedFilename);
```

Comment:
Could you use `static_cast`? I still ahve hope we can get this to compile on Ubuntu 16.04